xarray/test/test_dataarray.py

```diff
@@ -1244,6 +1244,53 @@ def test_groupby_first_and_last(self):
         expected = array  # should be a no-op
         self.assertDataArrayIdentical(expected, actual)
 
+    def make_groupby_multidim_example_array(self):
+        return DataArray([[0,1],[2,3]],
+                        coords={'lon': (['ny','nx'], [[30,40],[40,50]] ),
+                                'lat': (['ny','nx'], [[10,10],[20,20]] ),},
+                        dims=['ny','nx'])
```

Comment:
I'd also like to see this tested using floating point coordinates instead of integers.